pkg/operator/ceph/predicate.go

```diff
@@ -77,17 +78,28 @@ func predicateController(ctx context.Context, client client.Client) predicate.Fu
 						// We still want to reconcile the operator manager if the configmap is updated
 						return true
 					}
-				} else if s, ok := e.ObjectNew.(*v1.Secret); ok {
-					if s.Name == admissionControllerAppName {
-						logger.Debug("webhook secret updated, reloading the manager")
+				}
+			} else if s, ok := e.ObjectNew.(*v1.Secret); ok {
+				if s.Name == admissionControllerAppName {
+					logger.Debug("webhook secret updated, reloading the manager")
+					controller.ReloadManager()
+
+					// No need to ask for reconciliation since the context is going to be terminated when
+					// the signal is caught and the reconcile will run when the controller starts.
+					// If the admission controller secret is created or deleted we still need to reload and
+					// the webhook might be enabled or disabled
+					//
+					// The same goes the update, the secret changes we still need to reload the webhook
+					return false
+				}
+			} else if old, ok := e.ObjectOld.(*api.Certificate); ok {
+				if new, ok := e.ObjectNew.(*api.Certificate); ok {
+					if old.Name == certificateName && new.Name == certificateName {
```

Comment:
can we check it like `old.Name==new.Name`

Reply:
this condition covers that right?

Reply:
I mean 

> testing result when I deleted the certificate
> 
> ```
> srai@192 ~ (fix-webhook-rbac) $ kc delete certificates rook-admission-controller-cert 
> certificate.cert-manager.io "rook-admission-controller-cert" deleted
> ```
> 
> we can see operator reconciled and created the certificate again and webhook started
> 
> ```
> 2022-10-28 09:18:44.043205 I | operator: reloading operator's CRDs manager, cancelling all orchestrations!
> 2022-10-28 09:18:44.044457 I | op-mon: stopping monitoring of mons in namespace "rook-ceph"
> I1028 09:18:44.044633       1 manager.go:148] objectbucket.io/provisioner-manager "msg"="stopping provisioner" "name"="rook-ceph.ceph.rook.io/bucket" "reason"="context canceled"
> 2022-10-28 09:18:44.046045 I | ceph-cluster-controller: stopping monitoring of ceph status
> 2022-10-28 09:18:44.046250 I | op-osd: stopping monitoring of OSDs in namespace "rook-ceph"
> 2022-10-28 09:18:44.055284 I | operator: successfully started the controller-runtime manager
> 2022-10-28 09:18:44.060127 I | op-k8sutil: ROOK_CURRENT_NAMESPACE_ONLY="false" (env var)
> 2022-10-28 09:18:44.062922 I | operator: watching all namespaces for Ceph CRs
> 2022-10-28 09:18:44.063582 I | operator: setting up schemes
> 2022-10-28 09:18:44.069005 I | operator: setting up the controller-runtime manager
> 2022-10-28 09:18:44.674923 I | operator: Fetching webhook cert-manager-webhook to see if cert-manager is installed.
> 2022-10-28 09:18:44.679033 I | operator: Fetching Issuer rook-ceph/selfsigned-issuer.
> 2022-10-28 09:18:44.683678 I | operator: Fetching certificate rook-ceph/rook-admission-controller-cert.
> 2022-10-28 09:18:44.685419 I | operator: Creating certificate rook-ceph/rook-admission-controller-cert.
> 2022-10-28 09:18:44.707404 I | operator: waiting for webhook config "rook-admission-controller-cert" to be in ready status
> 2022-10-28 09:18:45.728800 I | operator: looking for admission webhook secret "rook-ceph-admission-controller"
> 2022-10-28 09:18:45.737013 I | operator: admission webhook secret "rook-ceph-admission-controller" found
> 2022-10-28 09:18:45.737042 I | operator: Creating webhook rook-ceph/rook-ceph-webhook.
> 2022-10-28 09:18:45.744861 I | operator: admission webhook already exists. updating it
> 2022-10-28 09:18:45.783340 I | operator: setting up admission webhooks
> 2022-10-28 09:18:45.783827 I | ceph-cluster-controller: successfully started
> ```

Is this log printed above the logs, `logger.Infof("webhook certificate %s deleted reloading the manager to enable the webhook server", certificateName)` ?

Reply:
Yaa it covers, 
I was thinking if we can use only this instead of two conditions.
But fine if you want  to keep it with the above flow